src/examples/java/io/nats/examples/NatsJsManageStreams.java

```diff
@@ -0,0 +1,143 @@
+                    // .maxConsumers(...)
+                    // .maxBytes(...)
+                    // .maxAge(...)
+                    // .maxMsgSize(...)
+                    .storageType(StreamConfiguration.StorageType.Memory)
+                    // .replicas(...)
+                    // .noAck(...)
+                    // .template(...)
+                    // .discardPolicy(...)
+                    .build();
+            StreamInfo streamInfo = jsm.addStream(streamConfig);
+            NatsJsUtils.printStreamInfo(streamInfo);
+
+            // 2. Update stream, in this case add a subject
+            // -  StreamConfiguration is immutable once created
+            // -  but the builder can help with that.
+            System.out.println("----------\n2. Update Stream 1");
+            streamConfig = StreamConfiguration.builder(streamInfo.getConfiguration())
+                    .addSubjects(SUBJECT2).build();
+            streamInfo = jsm.updateStream(streamConfig);
+            NatsJsUtils.printStreamInfo(streamInfo);
+
+            // 3. Create (add) another stream with 2 subjects
+            System.out.println("----------\n3. Configure And Add Stream 2");
+            streamConfig = StreamConfiguration.builder()
+                    .name(STREAM2)
+                    .storageType(StreamConfiguration.StorageType.Memory)
+                    .subjects(SUBJECT3, SUBJECT4)
+                    .build();
+            streamInfo = jsm.addStream(streamConfig);
+            NatsJsUtils.printStreamInfo(streamInfo);
+
+            // 4. Get information on streams
+            // 4.0 publish some message for more interesting stream state information
+            // -   SUBJECT1 is associated with STREAM1
+            // 4.1 getStreamInfo on a specific stream
+            // 4.2 get a list of all streams
+            // 4.3 get a list of StreamInfo's for all streams
+            System.out.println("----------\n4.1 getStreamInfo");
+            publish(nc, SUBJECT1, 5);
+            streamInfo = jsm.getStreamInfo(STREAM1);
+            NatsJsUtils.printStreamInfo(streamInfo);
+
+            System.out.println("----------\n4.2 getStreamNames");
+            List<String> streamNames = jsm.getStreamNames();
+            printObject(streamNames);
+
+            System.out.println("----------\n4.2 getStreamNames");
+            List<StreamInfo> streamInfos = jsm.getStreams();
+            NatsJsUtils.printStreamInfoList(streamInfos);
+
+            // 5. Purge a stream of it's messages
+            System.out.println("----------\n5. Purge stream");
+            PurgeResponse purgeResponse = jsm.purgeStream(STREAM1);
+            printObject(purgeResponse);
+
+            // 6.1 Delete a stream
+            // 6.2 getStreamInfo should exception
+            // 6.3 deleteStream should exception
+            // 6.4 purgeStream should exception
+            System.out.println("----------\n6. Delete stream");
+            jsm.deleteStream(STREAM2);
```

Comment:
I'm not sure the calls below are required for the example, the APIs are demonstrated above - I'd think the comment above noting that subsequent calls will throw an exception should suffice.

Reply:
fixed